File: code/ch01/1.3.2.two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> mapper = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      if (mapper.containsKey(target - nums[i])) {
        return new int[] {mapper.get(target - nums[i]), i};
      }
      mapper.put(nums[i], i);
    }
    return new int[] {};
  }
}

File: code/ch01/1.3.2.two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        mapper = {}
        for i in range(n):
            if target - nums[i] in mapper:
                return [mapper[target - nums[i]], i]
            else:
                mapper[nums[i]] = i

        return []
